core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/ExcelInputFormat.java

```diff
@@ -0,0 +1,251 @@
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+import org.apache.hadoop.mapreduce.lib.input.TextInputFormat;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.usermodel.WorkbookFactory;
+import org.apache.poi.ss.util.CellReference;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+
+/**
+ * {@link ExcelInputFormat} is {@link TextInputFormat} implementation for reading Excel files.
+ *
+ * The {@link ExcelInputFormat.ExcelRecordReader} reads a given sheet, and within a sheet reads
+ * all columns and all rows.
+ */
+public class ExcelInputFormat extends TextInputFormat {
+
+  public static final String SHEET_NAME = "sheetName";
+  public static final String RE_PROCESS = "reprocess";
+  public static final String SHEET_NO = "sheetNo";
+  public static final String COLUMN_LIST = "columnList";
+  public static final String SKIP_FIRST_ROW = "skipFirstRow";
+  public static final String TERMINATE_IF_EMPTY_ROW = "terminateIfEmptyRow";
+  public static final String ROWS_LIMIT = "rowsLimit";
+  public static final String IF_ERROR_RECORD = "IfErrorRecord";
+  public static final String PROCESSED_FILES = "processedFiles";
+  public static final String FILE_PATTERN = "filePattern";
+
+  @Override
+  public RecordReader<LongWritable, Text> createRecordReader(InputSplit split, TaskAttemptContext context) {
+    return new ExcelRecordReader();
+  }
+
+  public static void setConfigurations(Job job, String filePattern, String sheetName, boolean reprocess,
+                                       int sheetNo, String columnList, boolean skipFirstRow, String terminateIfEmptyRow,
+                                       String rowLimit, String ifErrorRecord, String processedFiles) {
+
+    Configuration configuration = job.getConfiguration();
+    configuration.set(FILE_PATTERN, filePattern);
+    configuration.set(SHEET_NAME, sheetName);
+    configuration.setBoolean(RE_PROCESS, reprocess);
+    configuration.setInt(SHEET_NO, sheetNo);
+    configuration.set(COLUMN_LIST, columnList);
+    configuration.setBoolean(SKIP_FIRST_ROW, skipFirstRow);
+    configuration.set(TERMINATE_IF_EMPTY_ROW, terminateIfEmptyRow);
+
+    if (!Strings.isNullOrEmpty(rowLimit)) {
+      configuration.set(ROWS_LIMIT, rowLimit);
+    }
+
+    configuration.set(IF_ERROR_RECORD, ifErrorRecord);
+    configuration.set(PROCESSED_FILES, processedFiles);
+  }
+
+
+  /**
+   * Reads excel spread sheet, where the keys are the offset in the excel file and the text is the complete record.
+   */
+  public static class ExcelRecordReader extends RecordReader<LongWritable, Text> {
+
+    public static final String END = "END";
+    public static final String MID = "MID";
+    public static final String CELL_SEPERATION = "\t";
```

Comment:
Should we pick something that's not printable, but still on ASCII table ? The cell data might have a tab no ?